File: reverse_array.cpp
#include <iostream>
using namespace std;

void reverseArray(int arr[], int size) {
    int start = 0;
    int end = size - 1;
    
    while (start < end) {
        int temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
}

int main() {
    int array[] = {1, 2, 3, 4, 5};
    int size = sizeof(array) / sizeof(array[0]);
    
    cout << "Original array: ";
    for (int i = 0; i < size; i++) {
        cout << array[i] << " ";
    }
    
    reverseArray(array, size);
    
    cout << "\nReversed array: ";
    for (int i = 0; i < size; i++) {
        cout << array[i] << " ";
    }
    
    return 0;
}

File: ReverseArray.java
public class ReverseArray {
    public static void reverseArray(int[] arr) {
        int start = 0;
        int end = arr.length - 1;
        
        while (start < end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }

    public static void main(String[] args) {
        int[] array = {1, 2, 3, 4, 5};
        
        System.out.print("Original array: ");
        for (int num : array) {
            System.out.print(num + " ");
        }
        
        reverseArray(array);
        
        System.out.print("\nReversed array: ");
        for (int num : array) {
            System.out.print(num + " ");
        }
    }
}